lib/api/src/externals/function.rs

```diff
@@ -424,6 +424,7 @@ impl Function {
         func: &WasmFunctionDefinition,
         params: &[Val],
         results: &mut [Val],
+        trampoline_checked: bool,
```

Comment:
`fn call_wasm` now needs to be unsafe if we have a way to call it without checks.  Alternatively make a separate function.

We'll need to document in the `# Safety` section the correct way to call this function.